File: makefile/src/main/java/name/kropp/intellij/makefile/psi/MakefileTargetPresentation.java
package name.kropp.intellij.makefile.psi;

import com.intellij.navigation.ItemPresentation;
import name.kropp.intellij.makefile.MakefileFileTypeKt;

import javax.swing.*;

class MakefileTargetPresentation implements ItemPresentation {
    private MakefileTarget target;

    MakefileTargetPresentation(MakefileTarget target) {
        this.target = target;
    }

    @Override
    public String getPresentableText() {
        return target.getText();
    }

    @Override
    public String getLocationString() {
        return "in " + target.getContainingFile().getVirtualFile().getPresentableName();
    }

    @Override
    public Icon getIcon(boolean b) {
        return MakefileFileTypeKt.getMakefileTargetIcon();
    }
}

File: makefile/src/com/jetbrains/lang/makefile/psi/MakefileTargetPresentation.kt
package name.kropp.intellij.makefile.psi

import com.intellij.navigation.ItemPresentation
import name.kropp.intellij.makefile.MakefileTargetIcon

class MakefileTargetPresentation(private val target: MakefileTarget) : ItemPresentation {
  override fun getIcon(b: Boolean) = MakefileTargetIcon
  override fun getPresentableText() = target.text
  override fun getLocationString() = "in " + target.containingFile?.virtualFile?.presentableName
}